client/src/actions/books.js

```diff
@@ -0,0 +1,177 @@
+import axios from 'axios';
+import {
+  SET_BOOKS,
+  ADD_BOOK,
+  BOOK_FETCHED,
+  BOOK_UPDATED,
+  BOOK_DELETED,
+  BOOKS_SEARCHED } from './types';
+import { addFlashMessage } from './flashMessages';
+
+const API_VERSION = window.API_VERSION;
+
+export function booksSearched(result) { // eslint-disable-line require-jsdoc
+  return {
+    type: BOOKS_SEARCHED,
+    result
+  };
+}
+
+
+export function setBooks(books) { // eslint-disable-line require-jsdoc
+  return {
+    type: SET_BOOKS,
+    books
+  };
+}
+
+/**
+ * 
+ * 
+ * @export
+ * @param {any} bookId 
+ * @returns  {object} book action
+ */
+export function bookDeleted(bookId) {
+  return {
+    type: BOOK_DELETED,
+    bookId: bookId.id
+  };
+}
+
+/**
+ * 
+ * 
+ * @export
+ * @param {any} book 
+ * @returns  {object} book action
+ */
+export function bookFetched(book) {
+  return {
+    type: BOOK_FETCHED,
+    book
+  };
+}
+
+export function addBook(book) { // eslint-disable-line require-jsdoc
+  return {
+    type: ADD_BOOK,
+    book
+  };
+}
+
+
+/**
+ * 
+ * 
+ * @export
+ * @param {any} book 
+ * @returns  {object} bookUpdated Payload
+ */
+export function bookUpdated(book) {
+  return {
+    type: BOOK_UPDATED,
+    book
+  };
+}
+
+
+/**
+ * 
+ * 
+ * @export
+ * @returns  {func} promise
+ */
+export function getBooks() {
+  return dispatch => axios.get(`/api/${API_VERSION}/books`);
+}
+
+
+/**
+ * 
+ * 
+ * @export
+ * @param {any} data 
+ * @returns  {func} promise
+ */
+export function getBook(data) {
+  return dispatch => axios.get(`/api/${API_VERSION}/books/${data.id}`).then(
+    (book) => {
+      dispatch(bookFetched(book.data.message[0]));
+    },
+    errors => errors
+  );
+}
+
+export function saveBooks(data) { // eslint-disable-line require-jsdoc
+  return (dispatch) => {
+    const newData = { ...data, coverPhotoPath: '' };
+    return axios.post(`/api/${API_VERSION}/books`, newData);
+  };
+}
+
+/**
+ * 
+ * 
+ * @export
+ * @param {any} data 
+ * @returns {func} promise
+ */
+export const updateBookCover = data => dispatch => axios.put(`/api/${API_VERSION}/books/${data.id}?fields=coverPhotoPath`, data);
```

Comment:
One-liner is good but in some cases, it trades off readability. You want to fix that here.